[AArch64][DAG] Allow fptos/ui.sat to scalarized.

davemgreen opened the pull request:
We we previously running into problems with fp128 types and certain integer sizes.

Fixes an issue reported on #124984

alanzhao1 commented:
Confirmed this fixes the crash mentioned in https://github.com/llvm/llvm-project/pull/124984#issuecomment-2648061111. Thanks!

arsenm commented on llvm/lib/CodeGen/SelectionDAG/LegalizeVectorTypes.cpp:
```diff
@@ -882,6 +886,19 @@ SDValue DAGTypeLegalizer::ScalarizeVecOp_UnaryOp(SDNode *N) {
   return DAG.getNode(ISD::SCALAR_TO_VECTOR, SDLoc(N), N->getValueType(0), Op);
 }
 
+/// Same as ScalarizeVecOp_UnaryOp with an extra typesize operand.
```
I was going to say this is identical to ScalarizeVecRes_InregOp, but that appears to convert the type argument to the element type. Is it already the scalar type for these nodes? We probably should consistently just use the scalar type in all of these value type operands

davemgreen replied:
Yeah - the type sizes are scalar for vector operations, similar to assertzext/assertsext. In practice this is really just an extra operand that we don't need to touch while scalarizing. I can rename this to UnaryOpWithExtraInput to match the existing ScalarizeVecRes_UnaryOpWithExtraInput.

davemgreen committed:
[AArch64][DAG] Allow fptos/ui.sat to scalarized.

We we previously running into problems with fp128 types and certain integer
sizes.

davemgreen commented:
/cherry-pick bf7af2d12e3bb8c7bc322ed1c5bf4e9904ad409c

bgra8 commented:
thanks a lot for the quick fix @davemgreen !!
